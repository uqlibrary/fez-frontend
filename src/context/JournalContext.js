import React from 'react';

export default React.createContext({
    journalDetails: null,
    jnlDisplayType: null,
});
