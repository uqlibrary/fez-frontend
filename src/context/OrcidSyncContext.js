import { createContext } from 'react';

export const OrcidSyncContext = createContext({
    orcidSyncProps: {},
});

export default OrcidSyncContext;
