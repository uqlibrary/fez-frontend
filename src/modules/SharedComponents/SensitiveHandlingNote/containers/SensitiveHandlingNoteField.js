import React, { useState } from 'react';
import { NewGenericSelectField } from 'modules/SharedComponents/GenericSelectField';
import { TextField } from 'modules/SharedComponents/Toolbox/TextField';
import { SENSITIVE_HANDLING_NOTE_OTHER_TYPE, SENSITIVE_HANDLING_NOTE_TYPE } from 'config/general';
import { selectFields } from 'locale/selectFields';
import { Field } from 'redux-form/immutable';
import { useFormValuesContext } from 'context';
import { validation } from 'config';

export const isSensitiveHandlingNoteTypeOther = value => parseInt(value, 10) === SENSITIVE_HANDLING_NOTE_OTHER_TYPE;

export const SensitiveHandlingNoteField = props => {
    const { formValues } = useFormValuesContext();
    const [isOther, setIsOther] = useState(isSensitiveHandlingNoteTypeOther(formValues.sensitiveHandlingNote?.id));
    const handleSensitiveHandlingNoteIdChange = value => setIsOther(isSensitiveHandlingNoteTypeOther(value));
    return (
        <>
            <Field
                {...props}
                component={NewGenericSelectField}
                name={'filesSection.sensitiveHandlingNote.id'}
                textFieldId={'rek-sensitive-handling-note-id'}
                genericSelectFieldId="rek-sensitive-handling-note-id"
                itemsList={[{ value: '0', text: 'None' }, ...SENSITIVE_HANDLING_NOTE_TYPE]}
                onChange={handleSensitiveHandlingNoteIdChange}
                {...selectFields.sensitiveHandlingNoteType}
            />
            {isOther && (
                <Field
                    component={TextField}
                    label={'Sensitive Handling Note (Other)'}
                    name={'filesSection.sensitiveHandlingNote.other'}
                    textFieldId={'rek-sensitive-handling-note-other'}
                    fullWidth
                    multiline
                    minRows={6}
                    maxRows={6}
                    inputProps={{ maxLength: 65535 }}
                    validate={value => isOther && validation.required(value)}
                />
            )}
        </>
    );
};

export default SensitiveHandlingNoteField;
