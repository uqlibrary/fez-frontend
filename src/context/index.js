import { useContext } from 'react';

import AccountContext from './AccountContext';
import EditableContext from './EditableContext';
import FormErrorsContext from './FormErrorsContext';
import FormValuesContext from './FormValuesContext';
import LocallyStoredReducerContext from './LocallyStoredReducerContext';
import OrcidSyncContext from './OrcidSyncContext';
import RecordContext from './RecordContext';
import RecordsSelectorContext from './RecordsSelectorContext';
import TabbedContext from './TabbedContext';
import ScrollToSectionContext from './ScrollToSectionContext';
import JournalContext from './JournalContext';

export const useAccountContext = () => useContext(AccountContext);
export const useEditableContext = () => useContext(EditableContext);
export const useFormErrorsContext = () => useContext(FormErrorsContext);
export const useFormValuesContext = () => useContext(FormValuesContext);
export const useLocallyStoredReducerContext = () => useContext(LocallyStoredReducerContext);
export const useOrcidSyncContext = () => useContext(OrcidSyncContext);
export const useRecordContext = () => useContext(RecordContext);
export const useRecordsSelectorContext = () => useContext(RecordsSelectorContext);
export const useTabbedContext = () => useContext(TabbedContext);
export const useScrollToSectionContext = () => useContext(ScrollToSectionContext);
export const useJournalContext = () => useContext(JournalContext);

export {
    AccountContext,
    EditableContext,
    FormErrorsContext,
    FormValuesContext,
    LocallyStoredReducerContext,
    OrcidSyncContext,
    RecordContext,
    RecordsSelectorContext,
    ScrollToSectionContext,
    TabbedContext,
    JournalContext,
};
