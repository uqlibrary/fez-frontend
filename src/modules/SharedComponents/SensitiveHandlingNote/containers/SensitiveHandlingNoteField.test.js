import SensitiveHandlingNoteField from './SensitiveHandlingNoteField';

jest.mock('../../../../context');
import { useFormValuesContext } from 'context';

function setup(testProps = {}) {
    const props = {
        ...testProps,
    };

    return renderComponent(SensitiveHandlingNoteField, props);
}

describe('SensitiveHandlingNoteField', () => {
    beforeEach(() => {
        useFormValuesContext.mockImplementation(() => ({
            formValues: {},
        }));
    });

    afterEach(() => {
        useFormValuesContext.mockReset();
    });

    it('should render default view', () => {
        const render = setup({});
        expect(render.getRenderOutput()).toMatchSnapshot();
    });

    it('should render disabled view', () => {
        const render = setup({ disabled: true });
        expect(render.getRenderOutput()).toMatchSnapshot();
    });
});
