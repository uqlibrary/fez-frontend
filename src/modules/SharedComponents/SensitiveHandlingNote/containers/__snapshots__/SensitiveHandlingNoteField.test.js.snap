// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SensitiveHandlingNoteField should render default view 1`] = `
<React.Fragment>
  <Field
    ariaLabel="Please select sensitive handling note"
    component={
      {
        "$$typeof": Symbol(react.memo),
        "compare": null,
        "type": [Function],
      }
    }
    genericSelectFieldId="rek-sensitive-handling-note-id"
    itemsList={
      [
        {
          "text": "None",
          "value": "0",
        },
        {
          "text": "Indigenous/First Nations people should be aware that this output contains images, voices and/or names of deceased persons.",
          "value": 456855,
        },
        {
          "text": "Indigenous/First Nations people should be aware that this output is about women’s business.",
          "value": 456856,
        },
        {
          "text": "Indigenous/First Nations people should be aware that this output is about men’s business.",
          "value": 456857,
        },
        {
          "text": "Assessors should be aware that this output contains content related to any of the following: violence, family or domestic violence, self-harm, sexual assault, suicide, family child removal, refugee experiences, war survivor experiences or other traumatic experiences that may be distressing or harmful to some people.",
          "value": 456858,
        },
        {
          "text": "Assessors should be aware that this output contains content with explicit language, hate speech, nudity or sexuality, or drug use which may be confronting and potentially distressing to some people.",
          "value": 456859,
        },
        {
          "text": "Other",
          "value": 456860,
        },
      ]
    }
    label="Sensitive Handling Note"
    name="filesSection.sensitiveHandlingNote.id"
    onChange={[Function]}
    selectPrompt="Select a sensitive handling note"
    textFieldId="rek-sensitive-handling-note-id"
  />
</React.Fragment>
`;

exports[`SensitiveHandlingNoteField should render disabled view 1`] = `
<React.Fragment>
  <Field
    ariaLabel="Please select sensitive handling note"
    component={
      {
        "$$typeof": Symbol(react.memo),
        "compare": null,
        "type": [Function],
      }
    }
    disabled={true}
    genericSelectFieldId="rek-sensitive-handling-note-id"
    itemsList={
      [
        {
          "text": "None",
          "value": "0",
        },
        {
          "text": "Indigenous/First Nations people should be aware that this output contains images, voices and/or names of deceased persons.",
          "value": 456855,
        },
        {
          "text": "Indigenous/First Nations people should be aware that this output is about women’s business.",
          "value": 456856,
        },
        {
          "text": "Indigenous/First Nations people should be aware that this output is about men’s business.",
          "value": 456857,
        },
        {
          "text": "Assessors should be aware that this output contains content related to any of the following: violence, family or domestic violence, self-harm, sexual assault, suicide, family child removal, refugee experiences, war survivor experiences or other traumatic experiences that may be distressing or harmful to some people.",
          "value": 456858,
        },
        {
          "text": "Assessors should be aware that this output contains content with explicit language, hate speech, nudity or sexuality, or drug use which may be confronting and potentially distressing to some people.",
          "value": 456859,
        },
        {
          "text": "Other",
          "value": 456860,
        },
      ]
    }
    label="Sensitive Handling Note"
    name="filesSection.sensitiveHandlingNote.id"
    onChange={[Function]}
    selectPrompt="Select a sensitive handling note"
    textFieldId="rek-sensitive-handling-note-id"
  />
</React.Fragment>
`;
